shotover-proxy/src/transforms/cassandra/peers_rewrite.rs

```diff
@@ -17,67 +21,103 @@ impl CassandraPeersRewriteConfig {
     pub async fn get_transform(&self) -> Result<Transforms> {
         Ok(Transforms::CassandraPeersRewrite(CassandraPeersRewrite {
             port: self.port,
+            peer_table: FQName::new("system", "peers_v2"),
         }))
     }
 }
 
 #[derive(Clone)]
 pub struct CassandraPeersRewrite {
     port: u32,
+    peer_table: FQName,
 }
 
 #[async_trait]
 impl Transform for CassandraPeersRewrite {
     async fn transform<'a>(&'a mut self, mut message_wrapper: Wrapper<'a>) -> ChainResponse {
         // Find the indices of queries to system.peers & system.peers_v2
-        let system_peers = message_wrapper
+        // we need to know which columns in which CQL queries in which messages have system peers
+        let column_names: HashMap<usize, Vec<String>> = message_wrapper
             .messages
             .iter_mut()
             .enumerate()
-            .filter_map(|(i, m)| if is_system_peers(m) { Some(i) } else { None })
-            .collect::<Vec<_>>();
+            .filter_map(|(i, m)| {
+                let sys_peers = extract_native_port_column(&self.peer_table, m);
+                if sys_peers.is_empty() {
+                    None
+                } else {
+                    Some((i, sys_peers))
+                }
+            })
+            .collect();
 
         let mut response = message_wrapper.call_next_transform().await?;
 
-        for i in system_peers {
-            rewrite_port(&mut response[i], self.port);
+        for (idx, name_list) in column_names {
+            rewrite_port(&mut response[idx], &name_list, self.port);
         }
 
         Ok(response)
     }
 }
 
-fn is_system_peers(message: &mut Message) -> bool {
-    if let Some(Frame::Cassandra(_)) = message.frame() {
-        if let Some(namespace) = message.namespace() {
-            if namespace.len() > 1 {
-                return namespace[0] == "system" && namespace[1] == "peers_v2";
+/// determine if the message contains a SELECT from `system.peers_v2` that includes the `native_port` column
+/// return a list of column names (or their alias) for each `native_port`.
+fn extract_native_port_column(peer_table: &FQName, message: &mut Message) -> Vec<String> {
+    let mut result: Vec<String> = vec![];
+    if let Some(Frame::Cassandra(cassandra)) = message.frame() {
+        if let CassandraOperation::Query { query, .. } = &cassandra.operation {
+            for cql_statement in &query.statements {
+                let statement = &cql_statement.statement;
+                if let CassandraStatement::Select(select) = &statement {
+                    if peer_table.eq(&select.table_name) {
+                        select
+                            .columns
+                            .iter()
+                            .for_each(|select_element| match select_element {
```

Comment:
this is easier to read as a regular for loop `for select_element in &select.columns {`